src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
+        for (final Talk talk : talks.active()) {
+            final List<String> requests = talk.read().xpath("//request");
```

Comment:
I would move this "talk processing" block into its own method. Now this method is doing two things: iterating over talks and processing each talk. That's too much and the code is difficult to digest